components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java

```diff
@@ -508,13 +509,15 @@ protected void setAuthenticator(ApplicationAuthenticator authenticator) {
             localAuthenticatorConfig.setTags(getTags(authenticator));
             AuthenticatorConfig fileBasedConfig = getAuthenticatorConfig(authenticator.getName());
             localAuthenticatorConfig.setEnabled(fileBasedConfig.isEnabled());
+            localAuthenticatorConfig.setDefinedByType(DefinedByType.SYSTEM);
```

Comment:
If we are giving this property to authenticator then we should get it from there. else it's fine to say any authenticator registers as an osgi is a system authenticator

Reply:
Addressed with commit fe95d4b60aeb8557a6ac7cfe51ce2576ee94112c. Only Authenticator config has this property now.